contracts/SpokePool.sol

```diff
@@ -154,36 +154,53 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         _;
     }
 
+    modifier onlyAdmin() {
+        _requireAdminSender();
+        _;
+    }
+
     /**************************************
      *          ADMIN FUNCTIONS           *
      **************************************/
 
-    function _setCrossDomainAdmin(address newCrossDomainAdmin) internal {
-        require(newCrossDomainAdmin != address(0), "Bad bridge router address");
-        crossDomainAdmin = newCrossDomainAdmin;
-        emit SetXDomainAdmin(crossDomainAdmin);
+    function setCrossDomainAdmin(address newCrossDomainAdmin) public override onlyAdmin nonReentrant {
+        _setCrossDomainAdmin(newCrossDomainAdmin);
     }
 
-    function _setHubPool(address newHubPool) internal {
-        require(newHubPool != address(0), "Bad hub pool address");
-        hubPool = newHubPool;
-        emit SetHubPool(hubPool);
+    function setHubPool(address newHubPool) public override onlyAdmin nonReentrant {
+        _setHubPool(newHubPool);
     }
 
-    function _setEnableRoute(
+    function setEnableRoute(
         address originToken,
         uint256 destinationChainId,
         bool enabled
-    ) internal {
+    ) public override onlyAdmin nonReentrant {
+        _requireAdminSender();
```

Comment:
Why call both `_requireAdminSender()` and use the `onlyAdmin` modifier aren't these redundant?

Reply:
Yes! That's a mistake.

Reply:
Fixed.